### PR TITLE
Balder/remove version and avoid going between nbostrea and bytebuffer

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

File: document/src/vespa/document/update/fieldpathupdate.h [hunk 1]
```diff
@@ -84,7 +85,7 @@ class FieldPathUpdate : public vespalib::Cloneable,
      * @param buffer The byte buffer that contains the serialized object.
      * @param version The serialization version of the object to deserialize.
      */
-    virtual void deserialize(const DocumentTypeRepo& repo, const DataType& type, ByteBuffer& buffer);
+    virtual void deserialize(const DocumentTypeRepo& repo, const DataType& type, nbostream & stream);
```
Comment (vekterli) [inline document/src/vespa/document/update/fieldpathupdate.h]: Consider updating/removing outdated comments

File: document/src/vespa/document/update/valueupdate.cpp [hunk 2]
```diff
@@ -11,16 +11,17 @@ IMPLEMENT_IDENTIFIABLE_ABSTRACT(ValueUpdate, Identifiable);
...
-    buffer.getIntNetwork(classId);
+    stream >> classId;
 
     const Identifiable::RuntimeClass * rtc(Identifiable::classFromId(classId));
     if (rtc != nullptr) {
         std::unique_ptr<ValueUpdate> update(static_cast<ValueUpdate*>(rtc->create()));
         /// \todo TODO (was warning):  Updates are not versioned in serialization format. Will not work without altering it.
-        update->deserialize(repo, type, buffer, serializationVersion);
+        /// Should also use the serializer, not this deserilize into self.
+        update->deserialize(repo, type, stream);
```
Comment (vekterli) [inline document/src/vespa/document/update/valueupdate.cpp]: Typo: deserilize -> deserialize

File: document/src/vespa/document/update/fieldpathupdate.h [hunk 3]
```diff
@@ -87,8 +85,7 @@ class FieldPathUpdate : public vespalib::Cloneable,
      * @param buffer The byte buffer that contains the serialized object.
      * @param version The serialization version of the object to deserialize.
      */
-    virtual void deserialize(const DocumentTypeRepo& repo, const DataType& type,
-                             ByteBuffer& buffer, uint16_t version);
+    virtual void deserialize(const DocumentTypeRepo& repo, const DataType& type, nbostream & stream);
```
Comment (vekterli) [inline document/src/vespa/document/update/fieldpathupdate.h]: Consider updating/removing outdated comments

File: document/src/vespa/document/update/fieldupdate.h [hunk 4]
```diff
@@ -42,11 +43,10 @@ class FieldUpdate : public vespalib::Identifiable,
      * a byte buffer by deserializing all its content from the buffer.
      *
      * @param type A document type that describes the buffer content.
-     * @param buffer A byte buffer that contains a serialized field update.
+     * @param stream A stream that contains a serialized field update.
      * @param serializationVersion The serialization version the update was serialized with.
      */
-    FieldUpdate(const DocumentTypeRepo& repo, const DataType & type,
-                ByteBuffer& buffer, int16_t version);
+    FieldUpdate(const DocumentTypeRepo& repo, const DataType & type, nbostream & stream);
```
Comment (vekterli) [inline document/src/vespa/document/update/fieldupdate.h]: Consider removing outdated comments (`serializationVersion`)

File: document/src/vespa/document/update/fieldupdate.h [hunk 5]
```diff
@@ -91,10 +91,8 @@ class FieldUpdate : public vespalib::Identifiable,
      *
      * @param type A document type that describes the buffer content.
      * @param buffer The byte buffer that contains the serialized update object.
-     * @param serializationVersion The serialization version the update was serialized with.
      */
-    void deserialize(const DocumentTypeRepo& repo, const DocumentType& type,
-                     ByteBuffer& buffer, int16_t serializationVersion);
+    void deserialize(const DocumentTypeRepo& repo, const DocumentType& type, nbostream& stream);
```
Comment (vekterli) [inline document/src/vespa/document/update/fieldupdate.h]: Consider updating/removing outdated comments (`buffer`)

File: document/src/vespa/document/update/valueupdate.h [hunk 6]
```diff
@@ -88,11 +87,8 @@ class ValueUpdate : public vespalib::Identifiable,
      *
      * @param type A data type that describes the content of the buffer.
      * @param buffer The byte buffer that contains the serialized update object.
-     * @param version The serialization version of the object to deserialize.
      */
-    virtual void deserialize(const DocumentTypeRepo& repo,
-                             const DataType& type,
-                             ByteBuffer& buffer, uint16_t version) = 0;
+    virtual void deserialize(const DocumentTypeRepo& repo, const DataType& type, nbostream & stream) = 0;
```
Comment (vekterli) [inline document/src/vespa/document/update/valueupdate.h]: Consider updating/removing outdated comments (`buffer`)

**Review by vekterli — APPROVED** [review]
:+1: just minor cleanup 🚿. Most of the existing Doxygen-style comments on functions in this code are in the "fluff stating the obvious" category, so won't be missed if they are removed wholesale.

File: document/src/vespa/document/update/fieldpathupdate.h (hunk 1 above)
Comment (baldersheim) [reply document/src/vespa/document/update/fieldpathupdate.h]: Removed

**Comment by baldersheim** [comment]
All comments addressed.

**Pull request merged**